presto-product-tests/src/main/java/io/prestosql/tests/hive/ACIDTestHelper.java

```diff
@@ -0,0 +1,101 @@
+
+import java.net.URI;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.UUID;
+
+import static io.prestosql.tempto.assertions.QueryAssert.Row.row;
+import static io.prestosql.tempto.assertions.QueryAssert.assertThat;
+
+public final class ACIDTestHelper
+{
+    private static final String LOCALHOST = "localhost";
+
+    private ACIDTestHelper()
+    {
+    }
+
+    // This simulates a aborted transaction which leaves behind a file in a partition of a table as follows:
+    // 1. Open Txn
+    // 2. Rollback Txn
+    // 3. Create a new table with location as the given partition location + delta directory for the rolled back transaction
+    // 4. Insert something to it which will create a file for this table
+    public static void simulateAbortedHiveTransaction(String database, String tableName, String partitionSpec)
+            throws TException
+    {
+        ThriftHiveMetastoreClient client = createMetastoreClient();
+        try {
+            // Step 1
+            long transaction = client.openTransaction("test");
+
+            AllocateTableWriteIdsRequest allocateTableWriteIdsRequest = new AllocateTableWriteIdsRequest(database, tableName);
+            allocateTableWriteIdsRequest.setTxnIds(Collections.singletonList(transaction));
+            long writeId = client.allocateTableWriteIdsBatchIntr(allocateTableWriteIdsRequest).get(0).getWriteId();
+
+            // Step 2
+            client.rollbackTransaction(transaction);
+
+            // Step 3
+            Table table = client.getTableWithCapabilities(database, tableName);
+            String tableLocation = table.getSd().getLocation();
+            String partitionLocation = tableLocation.endsWith("/") ? tableLocation + partitionSpec : tableLocation + "/" + partitionSpec;
+            String deltaLocation = partitionLocation + "/" + AcidUtils.deltaSubdir(writeId, writeId, 0);
+            QueryExecutor hiveQueryExecutor = ThreadLocalTestContextHolder.testContext().getDependency(QueryExecutor.class, "hive");
+
+            String tmpTableName = getNewTableName();
+            hiveQueryExecutor.executeQuery(String.format("CREATE EXTERNAL TABLE %s (col string) stored as ORC location '%s'", tmpTableName, deltaLocation));
+
+            // Step 4
+            hiveQueryExecutor.executeQuery(String.format("INSERT INTO TABLE %s SELECT 'a'", tmpTableName));
+            assertThat(hiveQueryExecutor.executeQuery(String.format("SELECT * FROM %s", tmpTableName))).containsOnly(row("a"));
+            hiveQueryExecutor.executeQuery(String.format("DROP TABLE %s", tmpTableName));
+        }
+        finally {
+            client.close();
+        }
+    }
+
+    private static String getNewTableName()
+    {
+        return "table_" + UUID.randomUUID().toString().replace('-', '_');
+    }
+
+    private static ThriftHiveMetastoreClient createMetastoreClient()
+            throws TException
+    {
+        URI metastore = URI.create("thrift://hadoop-master:9083");
```

Comment:
You should not hardcode hostname and port.
Inject values into test code via:
```
    @Inject(optional = true)
    @Named("databases.hive.metastore.host")
    private String metastoreHost;

    @Inject(optional = true)
    @Named("databases.hive.metastore.port")
    private int metastorePort;
```

Reply:
I tried, but it is not able to fetch the values defined in yml file. can you point out what am i missing?

Reply:
https://github.com/prestosql/presto/commit/0303aa63f94cc72f568f70738d195ed3e78e8385
works fine

Reply:
thanks for the help, incorporated the changes.